RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegFit.cc

```diff
@@ -0,0 +1,614 @@
+// ------------------------- //
+// --> GEMCSCSegFit.cc 
+// Created:  21.04.2015
+// --> Based on CSCSegFit.cc 
+// with last mod: 03.02.2015
+// as found in 750pre2 rel
+// ------------------------- //
+
+#include "RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegFit.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/MuonDetId/interface/CSCDetId.h"
+
+#include "Geometry/CSCGeometry/interface/CSCLayer.h"
+#include "Geometry/GEMGeometry/interface/GEMEtaPartition.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+
+void GEMCSCSegFit::fit(void) {
+  edm::LogVerbatim("GEMCSCSegFit") << "[GEMCSCSegFit::fit] - start the fitting fun :: nhits = "<<nhits();
+  if ( fitdone() ) return; // don't redo fit unnecessarily
+  short n = nhits();
+  switch ( n ) {
+  case 1:
+    edm::LogVerbatim("GEMCSCSegFit") << "[GEMCSCSegFit::fit] - cannot fit just 1 hit!!";
+    break;
+  case 2:
+    fit2();
+    break;
+  case 3:
+  case 4:
+  case 5:
+  case 6:
+  case 7:
+  case 8:
+    fitlsq();
+    break;
+  default:
+    edm::LogVerbatim("GEMCSCSegFit") << "[GEMCSCSegFit::fit] - cannot fit more than 6 hits!!";
```

Comment:
It looks like the code fits up to 8 hits -- why does this message say you can't fit more than 6?